packages/core/primitive/src/primitive.tsx

```diff
@@ -18,4 +14,37 @@ function composeEventHandlers<E>(
   };
 }
 
+/**
+ * Flush custom event dispatch
+ * https://github.com/radix-ui/primitives/pull/1378
+ *
+ * React batches *all* event handlers since version 18, this introduces certain considerations when using custom event types.
+ *
+ * Internally, React prioritises events in the following order:
+ *  - discrete
+ *  - continous
+ *  - default
+ *
+ * `discrete` is an  important distinction as updates within these events are applied immediately.
+ * React however, is not able to infer the priority of custom event types due to how they are detected internally.
+ * Because of this, it's possible for updates from custom events to be unexpectedly batched when
+ * dispatched by another `discrete` event.
+ *
+ * In order to ensure that updates from custom events are applied predictably, we need to manually flush the batch.
+ * This utility should be used when dispatching a custom event from within another `discrete` event, for example:
+ *
+ * dispatching a click event:
+ * target.dispatchEvent(new Event(‘click’))
```

Comment:
Wouldn't this work automatically given that React would find that native type?

Reply:
Yup, I was trying to show an example of when you would / wouldn't use this util

For example
- A known type 👎 
- A custom type called by a non-discrete event 👎 
- A custom type called by a discrete event 👍 

Perhaps I can make this clearer in the comment some how 🤔

Reply:
oh right!!! I understand now.
Yeah I think we should re-write this a bit then because when reading this:

> This utility should be used when dispatching a custom event from within another `discrete` event, for example: 

I understood that all these should use it.

Perhaps as simple as being explicit and saying here's when you shouldn't do it, show the 2 first examples. Here's when you should do it (last example) with the descriptions you provided it's even better:

>A known type 👎
A custom type called by a non-discrete event 👎
A custom type called by a discrete event 👍

Reply:
Hopefully this is a clearer now https://github.com/radix-ui/primitives/pull/1378/commits/9d35751ca74ca99444709bab9e4fc83e7b7c4e5d